elements/lisk-transactions/src/base_transaction.ts

```diff
@@ -277,9 +278,27 @@ export abstract class BaseTransaction {
 			);
 		}
 
+		const feeError = this.validateFee();
+
+		if (feeError) {
+			errors.push(feeError);
+		}
+
 		return createResponse(this.id, errors);
 	}
 
+	public validateFee(): TransactionError | undefined {
+		return !this.fee.eq((this.constructor as typeof BaseTransaction).FEE)
+			? new TransactionError(
+					`Invalid fee`,
+					this.id,
+					'.fee',
+					this.fee.toString(),
+					(this.constructor as typeof BaseTransaction).FEE.toString(),
+			  )
+			: undefined;
```

Comment:
Why is undefined needed here?

Reply:
@MaciejBaj because if there's no error it returns undefined and I saw this was being used for example `validateTransactionId` so I tried to use the same "pattern"